File: ch01/python/ants.py
MAX_L = 1000000
MAX_N = 1000000

# 標準入力から n, edge を取得
L = int(input())
n = int(input())
x = []
for i in range(n):
    x.append(int(input()))

min_time = 0
max_time = 0
for i in range(n):
    min_time = max([min_time, min(x[i], L - x[i])])
    max_time = max([max_time, max(x[i], L - x[i])])

print("min_time: ", min_time, ", max_time: ", max_time)

File: ch01/cpp/ants.cpp
#include <cstdio>
#include <algorithm>

const int MAX_L = 1000000;
const int MAX_N = 1000000;

int L, n;
int x[MAX_N];

void solve() {
  // 最小時間
  int min_time = 0;
  for (int i = 0; i < n; i++) {
    min_time = std::max(min_time, std::min(x[i], L - x[i]));
  }

  // 最大時間
  int max_time = 0;
  for (int i = 0; i < n; i++) {
    max_time = std::max(max_time, std::max(x[i], L - x[i]));
  }

  // 表示
  printf("min_time: %d, max_time: %d\n", min_time, max_time);
}

int main() {
  scanf("%d %d", &L, &n);
  for (int i = 0; i < n; i++) {
    scanf("%d", &x[i]);
  }

  solve();

  return 0;
}
